pkgs/development/python-modules/beanprice/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, beancount
+, buildPythonPackage
+, click
+, fetchFromGitHub
+, pytestCheckHook
+, python-dateutil
+, regex
+, requests
+, setuptools
+}:
+buildPythonPackage rec {
+  pname = "beanprice";
+  version = "1.2.1-unstable-2024-06-19";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "beancount";
+    repo = "beanprice";
+    rev = "e894c9182f4d16f9a46ccb87bdaeca1a7dede040";
+    hash = "sha256-l96W77gldE06Za8fj84LADGCqlYeWlHKvWQO+oLy1gI=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    beancount
+    python-dateutil
+    requests
+  ];
+
+  nativeCheckInputs = [
+    click
+    pytestCheckHook
+    regex
+  ];
+
+  pythonImportsCheck = [
+    "beancount"
+    "beanprice"
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/beancount/beanprice";
+    description = "Price quotes fetcher for Beancount";
+    longDescription = ''
```

Comment:
Is this program expected to be both a cmdline and a library?

Reply:
Don't think so. Converted to use buildPythonApplication instead of buildPythonPackage